cmd/ci-operator-checkconfig/main.go

```diff
@@ -18,11 +22,47 @@ func main() {
 		os.Exit(1)
 	}
 
+	seen := map[api.ImageStreamTagReference][]*config.Info{}
 	if err := config.OperateOnCIOperatorConfigDir(configDir, func(configuration *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
 		// validation is implicit, so we don't need to do anything
+		// but record the images we saw for future validation
+		if configuration.PromotionConfiguration == nil {
+			return nil
+		}
+		tags, _ := release.ToPromote(*configuration.PromotionConfiguration, configuration.Images, sets.NewString())
```

Comment:
Also need to include all RawSteps.  And there may be rename shenanigans in RawSteps that relabel images (like artifacts) that we probably need to negative test for (are you promoting something that doesn't exist).

I've caught people promoting `src` via `additionalImages` - that was never really intended as a use case.  We can allow it for now, but in the future I reserve the right to slap hands.

Reply:
I've seen people promote `src` this way in tests-like repositories to allow other repos to use some kind of shared tests where a build was not necessary... I thought that was exactly the intended use case? At least that's what we recommended to people in the past.

Reply:
@smarterclayton why do we need to check `RawSteps`? For what? Promotion will not promote them unless it's identified using the `ToPromote` logic, so...

Reply:
AdditionalImages can reference RawSteps output, so promote validation needs to check if the additional images actually exists (unless you think validation already covers that).

Reply:
As discussed on slack I'll add that validation next (additionalImages must reference a real name).

This is lGTM then